packages/react/src/components/DataTable/Table.tsx

```diff
@@ -226,7 +232,9 @@ export const Table = ({
     <div
       className={`${prefix}--data-table-content`}
       // eslint-disable-next-line jsx-a11y/no-noninteractive-tabindex
-      tabIndex={isScrollable ? 0 : undefined}>
+      tabIndex={
+        tabIndex === undefined ? (isScrollable ? 0 : undefined) : tabIndex
```

Comment:
```suggestion
      tabIndex={tabIndex ?? (isScrollable ? 0 : undefined)}
```

this feels slightly cleaner without having to nest ternarys